pkg/util/util.go

```diff
@@ -0,0 +1,31 @@
+// Package util houses various utilities that do not fit anywhere else.
+package util
+
+import (
+	"runtime/debug"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
+)
+
+// RecoverGoroutine is a wrapper to be used for calling new goroutines / using run.Group
+// which handles panics gracefully - prints the info out to the error log
+// and returns an error.
+func RecoverGoroutine(logger log.Logger, f func() error, extrafields ...interface{}) func() error {
+	return func() (err error) {
+		defer func() {
+			if p := recover(); p != nil {
```

Comment:
We should decide how to handle errors.. Preferably just pass `func() error` to have caller decide about this?